core/quorum_validation.go

```diff
@@ -2,15 +2,33 @@ package core
 
 import (
 	"bytes"
+	"fmt"
+	"sync"
 
 	ipfsnode "github.com/ipfs/go-ipfs-api"
 	"github.com/rubixchain/rubixgoplatform/block"
 	"github.com/rubixchain/rubixgoplatform/contract"
+	"github.com/rubixchain/rubixgoplatform/core/wallet"
 	"github.com/rubixchain/rubixgoplatform/did"
 	"github.com/rubixchain/rubixgoplatform/token"
 	"github.com/rubixchain/rubixgoplatform/util"
 )
 
+type TokenStateCheckResult struct {
+	Token                 string
+	Status                bool
+	Error                 error
+	Message               string
+	tokenIDTokenStateData string
+}
+
+const (
+	exhausted = true
```

Comment:
it doesn't make sense to have enum if the only two values are true and false.

Reply:
I suggested enum to have more than 2 statuses

Reply:
Exhausted sounds negative, and true being its value is confusing.

Reply:
If we have it as 
const (
    Exhausted Status = iota
    Free
)

(here Status becomes a type)
It would be easier to add more Statuses later

Refer - https://www.sohamkamani.com/golang/enums/#enum-types